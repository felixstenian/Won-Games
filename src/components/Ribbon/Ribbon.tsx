import * as Style from './styles'

export type RibbonColors = 'primary' | 'secondary'
export type RibbonSizes = 'normal' | 'small'

export type RibbonProps = {
  children: React.ReactNode
  color?: RibbonColors
  size?: RibbonSizes
}

const Ribbon = ({
  children,
  color = 'primary',
  size = 'normal'
}: RibbonProps) => (
  <Style.Wrapper color={color} size={size}>
    {children}
  </Style.Wrapper>
)

export default Ribbon
