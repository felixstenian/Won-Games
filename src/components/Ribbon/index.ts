export { default } from './Ribbon'
export type { RibbonProps, RibbonColors, RibbonSizes } from './Ribbon'
