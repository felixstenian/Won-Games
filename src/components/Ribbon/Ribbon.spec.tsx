import { screen } from '@testing-library/react'

import Ribbon from '.'
import { renderWithTheme } from 'utils'

describe('<Ribbon />', () => {
  it('1. should render the reading', () => {
    const { container } = renderWithTheme(
      <Ribbon>
        <h1>Ribbon</h1>
      </Ribbon>
    )

    expect(screen.getByRole('heading', { name: /Ribbon/i })).toBeInTheDocument()

    expect(container.firstChild).toMatchSnapshot()
  })
  it('2. should render the text correctly', () => {
    renderWithTheme(<Ribbon>Best seller</Ribbon>)

    expect(screen.getByText(/Best seller/i)).toBeInTheDocument()
  })
  it('3. should render with the primary color', () => {
    renderWithTheme(<Ribbon>Best seller</Ribbon>)

    expect(screen.getByText(/Best seller/i)).toHaveStyle({
      backgroundColor: '#F231A5'
    })
  })
  it('4. should render with the secondary color', () => {
    renderWithTheme(<Ribbon color="secondary">Best seller</Ribbon>)

    expect(screen.getByText(/Best seller/i)).toHaveStyle({
      backgroundColor: '#3CD3C1'
    })
  })
  it('5. should render with the normal size as default', () => {
    renderWithTheme(<Ribbon>Best seller</Ribbon>)

    expect(screen.getByText(/Best seller/i)).toHaveStyle({
      height: '3.6rem',
      fontSize: '1.4rem'
    })
  })
  it('6. should render with the small size', () => {
    renderWithTheme(<Ribbon size="small">Best seller</Ribbon>)

    expect(screen.getByText(/Best seller/i)).toHaveStyle({
      height: '2.6rem',
      fontSize: '1.2rem'
    })
  })
  it('7. should render with edge in normal size', () => {
    renderWithTheme(<Ribbon>Best seller</Ribbon>)

    const ribbon = screen.getByText(/Best seller/i)
    expect(ribbon).toHaveStyleRule('border-right-width', '2rem', {
      modifier: '::before'
    })
    expect(ribbon).toHaveStyleRule('border-top-width', '1rem', {
      modifier: '::before'
    })
    expect(ribbon).toHaveStyleRule('top', '3.6rem', {
      modifier: '::before'
    })
  })
})
it('8. should render with edge in small size', () => {
  renderWithTheme(<Ribbon size="small">Best seller</Ribbon>)

  const ribbon = screen.getByText(/Best seller/i)
  expect(ribbon).toHaveStyleRule('border-right-width', '1.5rem', {
    modifier: '::before'
  })
  expect(ribbon).toHaveStyleRule('border-top-width', '0.7rem', {
    modifier: '::before'
  })
  expect(ribbon).toHaveStyleRule('top', '2.6rem', {
    modifier: '::before'
  })
})
