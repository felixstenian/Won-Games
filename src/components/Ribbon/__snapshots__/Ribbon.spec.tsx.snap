// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Ribbon /> 1. should render the reading 1`] = `
.c0 {
  position: absolute;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  top: 2.4rem;
  font-weight: 600;
  color: #FAFAFA;
  background-color: #F231A5;
  font-size: 1.4rem;
  height: 3.6rem;
  padding: 0 2.4rem;
  right: -2rem;
}

.c0::before {
  border-left-color: #b20b6f;
  border-top-color: #b20b6f;
}

.c0::before {
  top: 3.6rem;
  border-top-width: 1rem;
  border-right-width: 2rem;
}

.c0::before {
  content: '';
  position: absolute;
  right: 0;
  border-style: solid;
  border-left-width: 0;
  border-right-color: transparent;
  border-bottom-color: transparent;
  border-bottom-width: 1rem;
}

<div
  class="c0"
  color="primary"
>
  <h1>
    Ribbon
  </h1>
</div>
`;
